src/blr/cli/tagfastq.py

```diff
@@ -119,22 +101,42 @@ def parse_corrected_barcodes(open_file):
     return corrected_barcodes
 
 
-def parse_raw_barcodes(open_file):
-    """
-    Parse FASTA/FASTQ containing barcode sequences and return a dictionary with entry headers
-    pointing to a raw barcodes sequence.
-    :param open_file: dnaio odject.
-    :return: dict: entry headers pointing to a raw barcodes sequence
-    """
-    for barcode in tqdm(open_file, desc="Raw barcodes processed"):
-        header, _ = barcode.name.split(maxsplit=1)
-        yield {header: barcode.sequence}
+class BarcodeReader:
+    def __init__(self, filename):
+        self.cache = dict()
+        self.filename = filename
+        self.barcodes = self.parse()
```

Comment:
To mark these attributes as "private", you should prefix them with an underscore (`_cache`, etc.).

Reply:
It’s a bit unusual that the constructor leaves the object in a half-initialized state (`self.open_file` is set in `__enter__`). You should open the input file already here. Then, if you add a `close()` method as well, it is possible to close the `BarcodeReader` without using it as a context manager.

Reply:
And I think it’s enough to call the file `_file`. The expectation would be anyway that it is open.

Reply:
Thanks for the comments! I implemented the changes in the new commit.